### PR TITLE
[WIP] WP Dev Lib setup with Travis and architecture refactoring

### PR DESCRIPTION
**Pull request by ThierryA**

### CODE REVIEW

**Comment by hellofromtonya** [comment]
Hey @ThierryA, we're getting an error on the TravisCI builds:

```
ERROR: Referenced sniff "PHPCompatibility" does not exist
```

https://travis-ci.org/Getbeans/Beans/jobs/309656242

File: composer.json [hunk 1]
```diff
@@ -32,7 +46,8 @@
     "wp-coding-standards/wpcs": "^0.14.0",
     "phpunit/phpunit": "~5.7.9",
     "brain/monkey": "^2.0",
-    "mikey179/vfsStream": "^1.6"
+    "mikey179/vfsStream": "^1.6",
+    "xwp/wp-dev-lib": "^1.0.1"
```
Comment (hellofromtonya) [inline composer.json]: We also need to change the script for `test-integration` to call the right configuration file:

Change:
```
"test-integration": "vendor/bin/phpunit --testsuite integration"
```

to:
```
"test-integration": "vendor/bin/phpunit --testsuite integration --configuration tests/phpunit/integration/phpunit.xml.dist"
```

Comment (ThierryA) [reply composer.json]: Sure, thanks Tonya. This is WIP and I am literally about to take of back to Switzerland from WCUS but I should be able to get this 100% finish in the course of this week. I will make sure to address the comments above. This setup will be re-usable for sure the Beans site with the exception of being in the WP root file etc. which will need some small adjustments.

Comment (ThierryA) [reply composer.json]: Sure, thanks Tonya. This is WIP and I am literally about to take of back to Switzerland from WCUS but I should be able to get this 100% finish in the course of this week. I will make sure to address the comments above. This setup will be re-usable for sure the Beans site with the exception of being in the WP root file etc. which will need some small adjustments.

**Comment by hellofromtonya** [comment]
@ThierryA Changes are approved and all tests pass.  Go ahead and merge it when you're happy with this PR.  Great job BTW!

**Comment by ThierryA** [comment]
Thanks @hellofromtonya 👍 I am still working on it and will let you know when it is 100% done so that you can do a finale sanity check.

**Comment by ThierryA** [comment]
@hellofromtonya here is a summary of how the dev setup work for the wiki. The new setup runs multiple things.

**1. WPCS:** note that it runs WPCS for the diff changes made which is a strict requirement at this stage since some legacy code are not yet compliant.
**2. Unit Test:** these runs outside of a WP environment and have their own config file. It can be manually ran using `composer test-unit` at the theme root.
**3. Integration tests:** these must be ran in a WP environment using `composer test-unit` at the theme root  (example: ssh in a VVV in local dev env).

## Pre-commit hook
The pre-commit hook runs the exact same build as on Travis to speed development (it only runs integrations test if the theme is in a VVV or Docker environment which will be automatically booted by the pre-commit hook). To install it, simply use `./install-pre-commit-hook.sh` at the theme root. 

It is sometimes needed to push some WIP work which may not pass the build. To prevent the pre-commit hook from running in order to allow WIP work to be committed, the `--no-verify` parameter must be added (ex: `git commit -m --no-verify  "Commit message"`).

**Note that they bash scripts are used so it is not compatible with Windows. Windows users may use it in a VM environment such as VVV.**

## Todos
- get the pre-commit hook to work with local (by flywheel)
- add eslint setup
- (re-)add PHPunit coverage correctly

**Pull request merged**